recipes/mgs/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MgsConan(ConanFile):
+    name = "mgs"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "C++14 codec library"
+    homepage = "https://theodelrieu.github.io/mgs-docs"
+    version = "0.1.2"
+    license = "MIT"
+    topics = ("codec", "base64", "base32", "base16")
+    settings = "compiler"
+    _source_subfolder = "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "Visual Studio": "14",
+            "clang": "3.4",
+            "apple-clang": "3.4",
+        }
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version:
+            if tools.Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration("mgs requires C++14, which your compiler does not fully support.")
+        else:
+            self.output.warn("mgs requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name, self._source_subfolder)
+
+    def package(self):
+        self.copy("*", src=os.path.join(self._source_subfolder, "include"), dst="include")
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "mgs"
```

Comment:
What is the point of defining all these components? they currently don't define anything meaningful/usefull AFAIU

Reply:
In the long run there might be several packages for each module, since you might only need a few codecs instead of everything. Although it's only a matter of separating things right now.

Reply:
the thing is that it does not separate anything right now. A consumer referencing only the `meta` package in a cmake file can use all the other components too right ?
Wouldn't it be better if each component had its own include path ?

Reply:
>A consumer referencing only the meta package in a cmake file can use all the other components too right ?

You're right... Maybe all of this is overkill for now? I don't know if creating separate packages is the way to go here :/

Reply:
if all the components sources are expected to stay in the same repo, and always have the same version, I guess it would not be desirable to make a package for each one

Reply:
that said you can still have a separate include path for each component, by setting `self.cpp_info.components["foo"].includedirs = [os.path.join('include", "foo")]`

Reply:
>that said you can still have a separate include path for each component

That wouldn't work since every include path is of the form: `<mgs/component/stuff.hpp>`.